File: training/project_euler/p001_sum_multiples/p001_sum_multiples.ml
(* https://projecteuler.net/problem=1 *)

let is_multiple i j =
  i != 0 && i mod j = 0

let sum n =
  let rec aux i acc =
    if i < n then
      if is_multiple i 3 || is_multiple i 5 then
        aux (i + 1) (acc + i)
      else
        aux (i + 1) acc
    else
      acc
  in aux 3 0

let () =
  print_int (sum 1000);
  print_newline ()

File: training/project_euler/p001_sum_multiples/p001_sum_multiples.go
// https://projecteuler.net/problem=1

package main

func is_multiple(i, j int) bool {
	return (i != 0) && ((i % j) == 0)
}

func sum(n int) int {
	acc := 0
	for i := 3; i < n; i++ {
		if is_multiple(i, 3) || is_multiple(i, 5) {
			acc += i
		}
	}
	return acc
}

func main() {
	print(sum(1000))
}
